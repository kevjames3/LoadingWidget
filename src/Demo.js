import "./styles/App.css";
import SpinnerWidget from "./components/SpinnerWidget/SpinnerWidget";
import SpinnerWidgetController from "./components/SpinnerWidgetController/SpinnerWidgetController";
import React, { Component } from "react";

class Demo extends Component {
  constructor() {
    super();
    this.state = {
      value: 0,
      spinning: true,
    };
  }

  onSpinningChange = (shouldSpin) => {
    console.log(shouldSpin);
    this.setState({ spinning: shouldSpin });
  };
  onValueChange = (newVal) => {
    this.setState({ value: newVal });
  };

  render() {
    return (
      <React.Fragment>
        <SpinnerWidget
          value={this.state.value}
          spinning={this.state.spinning}
        />
        <hr />
        <SpinnerWidgetController
          spinning={this.state.spinning}
          value={this.state.value}
          onSpinningChange={this.onSpinningChange}
          onValueChange={this.onValueChange}
        />
      </React.Fragment>
    );
  }
}

export default Demo;
