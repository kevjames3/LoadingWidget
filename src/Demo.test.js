import React from "react";
import ReactDOM from "react-dom";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";
import Demo from "./Demo";

describe("Demo", () => {
  let container;

  beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
  });

  afterEach(() => {
    document.body.removeChild(container);
    container = null;
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Demo />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("start/stops the widget from spinning when I click the buttons", () => {
    act(() => {
      ReactDOM.render(<Demo />, container);
    });

    const stopButton = container.querySelector(".stop-spinning");
    const startButton = container.querySelector(".start-spinning");
    expect(container.querySelector("svg.animate")).toBeTruthy();

    act(() => {
      stopButton.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });

    expect(container.querySelector("svg.pause-animate")).toBeTruthy();

    act(() => {
      startButton.dispatchEvent(new MouseEvent("click", { bubbles: true }));
    });

    expect(container.querySelector("svg.animate")).toBeTruthy();
  });

  it("updates the number correctly", () => {
    act(() => {
      ReactDOM.render(<Demo />, container);
    });

    let input = container.querySelector(".percentage-input input");
    let displayedNumber = container.querySelector(".spinner_widget div.status");

    expect(input).toHaveValue(0);
    expect(displayedNumber.textContent).toBe("0%");

    act(() => {
      input.value = 25;
      input.dispatchEvent(new MouseEvent("input", { bubbles: true }));
    });

    expect(input).toHaveValue(25);
    expect(displayedNumber.textContent).toBe("25%");
  });
});
