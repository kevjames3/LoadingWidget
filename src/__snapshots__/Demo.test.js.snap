// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Demo renders correctly 1`] = `
Array [
  <div
    className="spinner-widget"
  >
    <svg
      className="animate"
      height="100%"
      preserveAspectRatio="none"
      viewBox="0 0 100 100"
      width="100%"
    >
      <circle
        className="outer-circle"
        cx="50"
        cy="50"
        r="25"
        style={
          Object {
            "strokeDasharray": "0 158",
          }
        }
      />
      <circle
        className="inner_circle"
        cx="50"
        cy="50"
        r="45"
      />
    </svg>
    <div
      className="status"
    >
      0
      %
    </div>
  </div>,
  <hr />,
  <div
    className="spinner-widget-controls"
  >
    <div
      className="button-controls"
    >
      <button
        className="start-spinning"
        onClick={[Function]}
      >
        Keep Spinning
      </button>
      <button
        className="stop-spinning"
        onClick={[Function]}
      >
        Stop Spinning
      </button>
    </div>
    <div
      className="percentage-input"
    >
      <label>
        Current percentage
      </label>
      <input
        defaultValue={0}
        max="100"
        min="0"
        onInput={[Function]}
        type="Number"
      />
    </div>
  </div>,
]
`;
